### DISCUSSION TITLE
Does faiss index on gpu error out for anyone else?

### DISCUSSION THREAD

**Discussion by ssharpe42** [huggingface/datasets#4195] [Q&A] [answered]
When trying to use the dataset indexing on a GPU, it fails below. 

I have `faiss-gpu==1.7.2` installed and `datasets==2.1.0`. 
```
File /opt/conda/envs/labeling/lib/python3.9/site-packages/datasets/search.py:320, in FaissIndex._faiss_index_to_device(index, device)
    317 if isinstance(device, int):
    318     # Positive integers are directly mapped to GPU ids
    319     if device > -1:
--> 320         faiss_res = faiss.StandardGpuResources()
    321         index = faiss.index_cpu_to_gpu(faiss_res, device, index)
    322     # And negative integers mean using all GPUs
    323     else:

NameError: name 'faiss' is not defined
```

If I don't specify a device it works on the cpu. Could this relate to the fact that faiss isn't imported inside `_faiss_index_to_device`? I see it imported in other functions but not here https://github.com/huggingface/datasets/blob/b564af7c40d8df6da482af27fe1766e873aa2dd2/src/datasets/search.py#L305

If I add the import, the error goes away.

**Comment by alvarobartt** [comment, marked as answer]
Hi @ssharpe42 this was just solved in 🤗 `datasets==2.2.0` available at [2.2.0](https://github.com/huggingface/datasets/releases/tag/2.2.0).

More information at https://github.com/huggingface/datasets/issues/4287 and PR at https://github.com/huggingface/datasets/pull/4288

**Reply by alvarobartt** [reply]
@mariosasko could we already mark this discussion as answered? Thanks 🤗

**Reply by ssharpe42** [reply]
Great thanks!
